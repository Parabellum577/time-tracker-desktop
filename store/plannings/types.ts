import { ActionType } from 'typesafe-actions'

import { IPlanning, IError, ITimeSummary } from '@types'
import * as actions from './actions'

export interface IReportColdPlannigs {
  planningID: number
  timeToreport: number
}

export type PlanningsAction = ActionType<typeof actions>

export interface IPlanningsState {
  readonly openPlannings: IPlanning[]
  readonly coldPlannings: IPlanning[]
  readonly closedPlannings: IPlanning[]
  readonly timeSummary: ITimeSummary
  readonly planningByTask?: IPlanning
  readonly startPlanningError?: IError
  readonly stopPlanningError?: IError
  readonly highlightedPlanning: string
}
