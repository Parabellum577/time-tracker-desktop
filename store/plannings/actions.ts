import {
  ICreatePlanningPayload,
  ISetExtraPayload,
  IStopPlanning,
  ISetStatusPayload,
  ISetPriorityPayload,
} from '@api-types'
import { createAction, createAsyncAction } from 'typesafe-actions'
import { IPlanning, IError, IReportColdPlannings, IReportManually, ITimeSummary } from '@types'

export const getOpenPlannings = createAsyncAction(
  'plannings/GET_OPEN_PLANNINGS_REQUEST',
  'plannings/GET_OPEN_PLANNINGS_SUCCESS',
  'plannings/GET_OPEN_PLANNINGS_FAILURE',
)<void, IPlanning[], IError>()

export const getColdPlannings = createAsyncAction(
  'plannings/GET_COLD_PLANNINGS_REQUEST',
  'plannings/GET_COLD_PLANNINGS_SUCCESS',
  'plannings/GET_COLD_PLANNINGS_FAILURE',
)<void, IPlanning[], IError>()

export const getClosedPlannings = createAsyncAction(
  'plannings/GET_CLOSED_PLANNINGS_REQUEST',
  'plannings/GET_CLOSED_PLANNINGS_SUCCESS',
  'plannings/GET_CLOSED_PLANNINGS_FAILURE',
)<void, IPlanning[], IError>()

export const startPlanning = createAsyncAction(
  'plannings/START_PLANNING_REQUEST',
  'plannings/START_PLANNING_SUCCESS',
  'plannings/START_PLANNING_FAILURE',
)<{ PlanningID: number; TracingID?: string }, number, IError>()

export const startPlanningSync = createAction('plannings/START_PLANNING_SYNC', action => (PlanningID: number) =>
  action(PlanningID),
)

export const stopPlanning = createAsyncAction(
  'plannings/STOP_PLANNING_REQUEST',
  'plannings/STOP_PLANNING_SUCCESS',
  'plannings/STOP_PLANNING_FAILURE',
)<IStopPlanning | void, void, IError>()

export const startColdPlanning = createAsyncAction(
  'plannings/START_COLD_PLANNING_REQUEST',
  'plannings/START_COLD_PLANNING_SUCCESS',
  'plannings/START_COLD_PLANNING_FAILURE',
)<number | void, number, IError>()

export const stopColdPlanning = createAsyncAction(
  'plannings/STOP_COLD_PLANNING_REQUEST',
  'plannings/STOP_COLD_PLANNING_SUCCESS',
  'plannings/STOP_COLD_PLANNING_FAILURE',
)<void, void, IError>()

export const createPlannings = createAsyncAction(
  'plannings/CREATE_PLANNING_REQUEST',
  'plannings/CREATE_PLANNING_SUCCESS',
  'plannings/CREATE_PLANNING_FAILURE',
)<Array<{ task: ICreatePlanningPayload; isStarted: boolean }>, void, IError>()

export const timeTickInActivePlanning = createAction(
  'plannings/TIME_TICK_IN_ACTIVE_PLANNING',
  action => (PlanningID: number, field: 'SpentOnline' | 'SpentOffline', additionalTime: number) =>
    action({ PlanningID, field, additionalTime }),
)

export const setSpentTime = createAction(
  'plannings/SET_SPENT_TIME',
  action => (PlanningID: number, SpentTime: number) => action({ PlanningID, SpentTime }),
)

export const setExtra = createAction('plannings/SET_EXTRA_PLANNING', action => (Planning: ISetExtraPayload) =>
  action({
    Estimation: Planning.Estimation,
    PlanningID: Planning.PlanningID,
    Reason: Planning.Reason,
  }),
)

export const coldPlanningRemap = createAction('plannings/COLD_PLANNING_REMAP', action => (From: number, To: number) =>
  action({ From, To }),
)

export const getTimeSummary = createAsyncAction(
  'plannings/TIME_SUMMARY_REQUEST',
  'plannings/TIME_SUMMARY_SUCCESS',
  'plannings/TIME_SUMMARY_FAILURE',
)<null, ITimeSummary, IError>()

export const closePlanning = createAsyncAction(
  'plannings/CLOSE_PLANNING_REQUEST',
  'plannings/CLOSE_PLANNING_SUCCESS',
  'plannings/CLOSE_PLANNING_FAILURE',
)<IPlanning, IPlanning, IError>()

export const addHighlightedPlanning = createAction('plannings/ADD_HIGHLIGHTED_PLANNING', action => (ID: string) =>
  action(ID),
)

export const removeHighlightedPlanning = createAction('plannings/REMOVE_HIGHLIGHTED_PLANNING', action => (ID: string) =>
  action(ID),
)

export const planningExpired = createAction('plannings/PLANNING_EXPIRED', action => (planning: IPlanning) =>
  action(planning),
)

export const closeColdPlanning = createAction('plannings/CHANGE_COLD_PLANNING_STATUS', action => (planningID: number) =>
  action(planningID),
)

export const reportColdPlannings = createAsyncAction(
  'plannings/REPORT_COLD_PLANNINGS_REQUEST',
  'plannings/REPORT_COLD_PLANNINGS_SUCCESS',
  'plannings/REPORT_COLD_PLANNINGS_FAILURE',
)<IReportColdPlannings, void, IError>()

export const reportManually = createAsyncAction(
  'plannings/REPORT_REPORT_MANUALLY_REQUEST',
  'plannings/REPORT_REPORT_MANUALLY_SUCCESS',
  'plannings/REPORT_REPORT_MANUALLY_FAILURE',
)<IReportManually, void, IError>()

export const clearPlannings = createAction('plannings/CLEAR_PLANNINGS', action => () => action())

export const setStatus = createAction('plannings/SET_STATUS', action => (payload: ISetStatusPayload) => action(payload))

export const setPriority = createAction('plannings/SET_PRIORITY', action => (payload: ISetPriorityPayload) =>
  action(payload),
)
