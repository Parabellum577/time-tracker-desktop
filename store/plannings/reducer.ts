import { getType } from 'typesafe-actions'

import * as actions from './actions'
import { PlanningsAction, IPlanningsState } from './types'
import { IPlanning } from '@services/types'
import { getFullSpentTime } from '@services/helpers'

const initialState: IPlanningsState = {
  openPlannings: [],
  closedPlannings: [],
  coldPlannings: [],
  highlightedPlanning: '',
  timeSummary: {
    sumTotalDay: 0,
    sumManualToday: 0,
    sumTotalMonth: 0,
    sumManualMonth: 0,
  },
}

export default (state: IPlanningsState = initialState, action: PlanningsAction) => {
  switch (action.type) {
    case getType(actions.getOpenPlannings.success):
      return { ...state, openPlannings: action.payload }
    case getType(actions.getOpenPlannings.failure):
      return { ...state, openPlannings: [] }
    case getType(actions.getColdPlannings.success):
      return { ...state, coldPlannings: action.payload }
    case getType(actions.getColdPlannings.failure):
      return { ...state, coldPlannings: [] }
    case getType(actions.getClosedPlannings.success):
      return { ...state, closedPlannings: action.payload }
    case getType(actions.getClosedPlannings.failure):
      return { ...state, closedPlannings: [] }
    case getType(actions.getTimeSummary.success):
      return { ...state, timeSummary: action.payload }
    case getType(actions.startPlanningSync):
    case getType(actions.startPlanning.success):
      return {
        ...state,
        openPlannings: state.openPlannings.map(p => (p.ID === action.payload ? { ...p, Active: true } : p)),
        coldPlannings: state.coldPlannings.map(p => (p.ID === action.payload ? { ...p, Active: true } : p)),
      }
    case getType(actions.startPlanning.failure):
      return {
        ...state,
        openPlannings: state.openPlannings.map(p => (p.Active ? { ...p, Active: false } : p)),
        coldPlannings: state.coldPlannings.map(p => (p.Active ? { ...p, Active: false } : p)),
        startPlanningError: action.payload,
      }

    case getType(actions.stopPlanning.success):
      return {
        ...state,
        openPlannings: state.openPlannings.map(p => (p.Active ? { ...p, Active: false } : p)),
        coldPlannings: state.coldPlannings.map(p => (p.Active ? { ...p, Active: false } : p)),
      }

    case getType(actions.stopPlanning.failure):
      return {
        ...state,
        openPlannings: state.openPlannings.map(p => (p.Active ? { ...p, Active: false } : p)),
        coldPlannings: state.coldPlannings.map(p => (p.Active ? { ...p, Active: false } : p)),
        stopPlanningError: action.payload,
      }

    case getType(actions.timeTickInActivePlanning):
      const { PlanningID, field, additionalTime } = action.payload
      return {
        ...state,
        openPlannings: state.openPlannings.map(p => {
          if (p.ID === PlanningID) {
            p = { ...p }
            p[field] += additionalTime
          }
          return p
        }),
        coldPlannings: state.coldPlannings.map(p => {
          if (p.ID === PlanningID) {
            p = { ...p }
            p[field] += additionalTime
          }
          return p
        }),
        timeSummary: {
          ...state.timeSummary,
          sumTotalDay: state.timeSummary.sumTotalDay + additionalTime,
          sumTotalMonth: state.timeSummary.sumTotalMonth + additionalTime,
        },
      }

    case getType(actions.setSpentTime):
      return {
        ...state,
        openPlannings: state.openPlannings.map(p => {
          if (p.ID === action.payload.PlanningID) {
            p = { ...p }
            p.SpentOnline = action.payload.SpentTime
            p.SpentOffline = 0
            p.SpentManual = 0
          }
          return p
        }),
        coldPlannings: state.coldPlannings.map(p => {
          if (p.ID === action.payload.PlanningID) {
            p = { ...p }
            p.SpentOnline = action.payload.SpentTime
            p.SpentOffline = 0
            p.SpentManual = 0
          }
          return p
        }),
      }

    case getType(actions.coldPlanningRemap):
      const { From, To } = action.payload

      return {
        ...state,
        coldPlannings: state.coldPlannings
          .map(p => {
            const startOfDayRemapping = new Date(From * 1000).setHours(0, 0, 0, 0)
            const startOfDayCurrent = new Date(p.CreatedAt * 1000).setHours(0, 0, 0, 0)
            if (startOfDayCurrent === startOfDayRemapping) {
              p = { ...p, SpentOnline: p.SpentOnline - (To - From) }
            }
            return p
          })
          .filter(p => {
            return getFullSpentTime(p) > 0 || p.Status === 'OPEN'
          }),
      }

    case getType(actions.closePlanning.success):
      const Planning = action.payload
      return {
        ...state,
        openPlannings: state.openPlannings.filter(planning => planning.ID !== Planning.ID),
        closedPlannings: [...state.closedPlannings, state.openPlannings.find(planning => planning.ID === Planning.ID)],
      }
    case getType(actions.setExtra):
      const planningIDSetExtra = action.payload.PlanningID
      const planningForEdit = state.openPlannings.find(planning => planning.ID === planningIDSetExtra)
      const modifiedPlanning = {
        ...planningForEdit,
        Estimation: action.payload.Estimation,
        LastReason: action.payload.Reason,
      }
      return {
        ...state,
        openPlannings: [
          ...state.openPlannings.filter(planning => planning.ID !== planningIDSetExtra),
          modifiedPlanning,
        ],
      }
    case getType(actions.addHighlightedPlanning):
      return {
        ...state,
        highlightedPlanning: action.payload,
      }
    case getType(actions.removeHighlightedPlanning):
      return {
        ...state,
        highlightedPlanning: '',
      }
    case getType(actions.closeColdPlanning):
      const planningsToClose: IPlanning = state.coldPlannings.find(p => p.ID === action.payload)
      planningsToClose.Status = 'CLOSED'
      return {
        ...state,
        coldPlannings: [...state.coldPlannings, planningsToClose],
      }
    case getType(actions.clearPlannings):
      return { ...initialState }
    default:
      return state
  }
}
