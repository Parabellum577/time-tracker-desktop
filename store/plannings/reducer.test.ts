import { ActionType, getType } from 'typesafe-actions'

import * as actions from './actions'
import { IPlanningsState } from './types'
import plannings from './reducer'

import { SIMPLE_PLANNING } from '@test-helpers/sample'

describe('[Store] Plannings Reducer', () => {
  it('[Success] [Snapshot] Should handle GetOpenPlannings', () => {
    const action: ActionType<typeof actions> = {
      type: getType(actions.getOpenPlannings.success),
      payload: [{ ...SIMPLE_PLANNING }],
    }

    const currentState: IPlanningsState = {
      openPlannings: [],
      coldPlannings: [],
      closedPlannings: [],
      highlightedPlanning: '',
      timeSummary: {
        sumTotalDay: 0,
        sumManualToday: 0,
        sumTotalMonth: 0,
        sumManualMonth: 0,
      },
    }

    expect(plannings(currentState, action)).toMatchSnapshot()
  })

  it('[Failure] Should handle GetOpenPlannings', () => {
    const action: ActionType<typeof actions> = {
      type: getType(actions.getOpenPlannings.failure),
      payload: {
        code: 666,
        message: 'SOME_ERROR',
      },
    }

    const currentState: IPlanningsState = {
      openPlannings: [{ ...SIMPLE_PLANNING }],
      coldPlannings: [],
      closedPlannings: [],
      highlightedPlanning: '',
      timeSummary: {
        sumTotalDay: 0,
        sumManualToday: 0,
        sumTotalMonth: 0,
        sumManualMonth: 0,
      },
    }

    expect(plannings(currentState, action)).toMatchSnapshot()
  })

  it('[Success] Should handle GetClosedPlannings', () => {
    const action: ActionType<typeof actions> = {
      type: getType(actions.getClosedPlannings.success),
      payload: [{ ...SIMPLE_PLANNING }],
    }

    const currentState: IPlanningsState = {
      openPlannings: [],
      coldPlannings: [],
      closedPlannings: [],
      highlightedPlanning: '',
      timeSummary: {
        sumTotalDay: 0,
        sumManualToday: 0,
        sumTotalMonth: 0,
        sumManualMonth: 0,
      },
    }

    expect(plannings(currentState, action)).toMatchSnapshot()
  })

  it('[Failure] Should handle GetClosedPlannings', () => {
    const action: ActionType<typeof actions> = {
      type: getType(actions.getClosedPlannings.failure),
      payload: {
        code: 666,
        message: 'SOME_ERROR',
      },
    }

    const currentState: IPlanningsState = {
      openPlannings: [],
      coldPlannings: [],
      closedPlannings: [{ ...SIMPLE_PLANNING }],
      highlightedPlanning: '',
      timeSummary: {
        sumTotalDay: 0,
        sumManualToday: 0,
        sumTotalMonth: 0,
        sumManualMonth: 0,
      },
    }

    expect(plannings(currentState, action)).toMatchSnapshot()
  })

  it('[Success] Should handle StartPlanning', () => {
    const action: ActionType<typeof actions> = {
      type: getType(actions.startPlanning.success),
      payload: 33,
    }

    const currentState: IPlanningsState = {
      openPlannings: [
        {
          ...SIMPLE_PLANNING,
          Active: false,
          ID: 33,
        },
        {
          ...SIMPLE_PLANNING,
          Active: false,
          ID: 44,
        },
      ],
      coldPlannings: [],
      closedPlannings: [],
      highlightedPlanning: '',
      timeSummary: {
        sumTotalDay: 0,
        sumManualToday: 0,
        sumTotalMonth: 0,
        sumManualMonth: 0,
      },
    }

    expect(plannings(currentState, action)).toMatchSnapshot()
  })

  it('[Failure] Should handle StartPlanning', () => {
    const action: ActionType<typeof actions> = {
      type: getType(actions.startPlanning.failure),
      payload: {
        code: 666,
        message: 'SOME_ERROR',
      },
    }

    const currentState: IPlanningsState = {
      openPlannings: [
        {
          ...SIMPLE_PLANNING,
          Active: true,
          ID: 33,
        },
        {
          ...SIMPLE_PLANNING,
          Active: false,
          ID: 44,
        },
      ],
      coldPlannings: [],
      closedPlannings: [],
      highlightedPlanning: '',
      timeSummary: {
        sumTotalDay: 0,
        sumManualToday: 0,
        sumTotalMonth: 0,
        sumManualMonth: 0,
      },
    }

    expect(plannings(currentState, action)).toMatchSnapshot()
  })

  it('[Success] Should handle StopPlanning', () => {
    const action: any = {
      type: getType(actions.stopPlanning.success),
    }

    const currentState: IPlanningsState = {
      openPlannings: [
        {
          ...SIMPLE_PLANNING,
          Active: true,
          ID: 33,
        },
        {
          ...SIMPLE_PLANNING,
          Active: false,
          ID: 44,
        },
      ],
      coldPlannings: [],
      closedPlannings: [],
      highlightedPlanning: '',
      timeSummary: {
        sumTotalDay: 0,
        sumManualToday: 0,
        sumTotalMonth: 0,
        sumManualMonth: 0,
      },
    }

    expect(plannings(currentState, action)).toMatchSnapshot()
  })

  it('[Failure] Should handle StopPlanning', () => {
    const action: ActionType<typeof actions> = {
      type: getType(actions.stopPlanning.failure),
      payload: {
        code: 666,
        message: 'SOME_ERROR',
      },
    }

    const currentState: IPlanningsState = {
      openPlannings: [
        {
          ...SIMPLE_PLANNING,
          Active: true,
          ID: 33,
        },
        {
          ...SIMPLE_PLANNING,
          Active: false,
          ID: 44,
        },
      ],
      coldPlannings: [],
      closedPlannings: [],
      highlightedPlanning: '',
      timeSummary: {
        sumTotalDay: 0,
        sumManualToday: 0,
        sumTotalMonth: 0,
        sumManualMonth: 0,
      },
    }

    expect(plannings(currentState, action)).toMatchSnapshot()
  })

  it('[SyncSuccess] Should handle TimeTickInActivePlanning', () => {
    const action: ActionType<typeof actions> = {
      type: getType(actions.timeTickInActivePlanning),
      payload: {
        PlanningID: 33,
        field: 'SpentOnline',
        additionalTime: 1.001,
      },
    }

    const currentState: IPlanningsState = {
      openPlannings: [
        {
          ...SIMPLE_PLANNING,
          Active: true,
          ID: 33,
          SpentOnline: 200,
        },
        {
          ...SIMPLE_PLANNING,
          Active: false,
          ID: 44,
        },
      ],
      coldPlannings: [],
      closedPlannings: [],
      highlightedPlanning: '',
      timeSummary: {
        sumTotalDay: 0,
        sumManualToday: 0,
        sumTotalMonth: 0,
        sumManualMonth: 0,
      },
    }

    expect(plannings(currentState, action)).toMatchSnapshot()
  })
})
