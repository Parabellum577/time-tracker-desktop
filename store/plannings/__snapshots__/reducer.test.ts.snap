// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Store] Plannings Reducer [Failure] Should handle GetClosedPlannings 1`] = `
Object {
  "closedPlannings": Array [],
  "coldPlannings": Array [],
  "highlightedPlanning": "",
  "openPlannings": Array [],
  "timeSummary": Object {
    "sumManualMonth": 0,
    "sumManualToday": 0,
    "sumTotalDay": 0,
    "sumTotalMonth": 0,
  },
}
`;

exports[`[Store] Plannings Reducer [Failure] Should handle GetOpenPlannings 1`] = `
Object {
  "closedPlannings": Array [],
  "coldPlannings": Array [],
  "highlightedPlanning": "",
  "openPlannings": Array [],
  "timeSummary": Object {
    "sumManualMonth": 0,
    "sumManualToday": 0,
    "sumTotalDay": 0,
    "sumTotalMonth": 0,
  },
}
`;

exports[`[Store] Plannings Reducer [Failure] Should handle StartPlanning 1`] = `
Object {
  "closedPlannings": Array [],
  "coldPlannings": Array [],
  "highlightedPlanning": "",
  "openPlannings": Array [
    Object {
      "Active": false,
      "ActivityID": 0,
      "CloseComment": "string",
      "ClosedAt": 0,
      "ColdPlanningID": 0,
      "CreatedAt": 0,
      "Estimation": 0,
      "ID": 33,
      "IsColdPlanning": true,
      "IsManual": true,
      "IssueActivities": null,
      "IssueDone": 0,
      "IssueDueDate": 0,
      "IssueEstimation": 0,
      "IssueID": "string",
      "IssuePriorities": null,
      "IssueSpent": 0,
      "IssueStatuses": null,
      "IssueTitle": "string",
      "IssueTypes": null,
      "IssueURL": "string",
      "OpenComment": "string",
      "Outdated": true,
      "PlanningDone": true,
      "ProjectID": "string",
      "Reported": 0,
      "SpentManual": 0,
      "SpentOffline": 0,
      "SpentOnline": 0,
      "StartedBy": "string",
      "Status": "OPEN",
      "TrackerID": 0,
    },
    Object {
      "Active": false,
      "ActivityID": 0,
      "CloseComment": "string",
      "ClosedAt": 0,
      "ColdPlanningID": 0,
      "CreatedAt": 0,
      "Estimation": 0,
      "ID": 44,
      "IsColdPlanning": true,
      "IsManual": true,
      "IssueActivities": null,
      "IssueDone": 0,
      "IssueDueDate": 0,
      "IssueEstimation": 0,
      "IssueID": "string",
      "IssuePriorities": null,
      "IssueSpent": 0,
      "IssueStatuses": null,
      "IssueTitle": "string",
      "IssueTypes": null,
      "IssueURL": "string",
      "OpenComment": "string",
      "Outdated": true,
      "PlanningDone": true,
      "ProjectID": "string",
      "Reported": 0,
      "SpentManual": 0,
      "SpentOffline": 0,
      "SpentOnline": 0,
      "StartedBy": "string",
      "Status": "OPEN",
      "TrackerID": 0,
    },
  ],
  "startPlanningError": Object {
    "code": 666,
    "message": "SOME_ERROR",
  },
  "timeSummary": Object {
    "sumManualMonth": 0,
    "sumManualToday": 0,
    "sumTotalDay": 0,
    "sumTotalMonth": 0,
  },
}
`;

exports[`[Store] Plannings Reducer [Failure] Should handle StopPlanning 1`] = `
Object {
  "closedPlannings": Array [],
  "coldPlannings": Array [],
  "highlightedPlanning": "",
  "openPlannings": Array [
    Object {
      "Active": false,
      "ActivityID": 0,
      "CloseComment": "string",
      "ClosedAt": 0,
      "ColdPlanningID": 0,
      "CreatedAt": 0,
      "Estimation": 0,
      "ID": 33,
      "IsColdPlanning": true,
      "IsManual": true,
      "IssueActivities": null,
      "IssueDone": 0,
      "IssueDueDate": 0,
      "IssueEstimation": 0,
      "IssueID": "string",
      "IssuePriorities": null,
      "IssueSpent": 0,
      "IssueStatuses": null,
      "IssueTitle": "string",
      "IssueTypes": null,
      "IssueURL": "string",
      "OpenComment": "string",
      "Outdated": true,
      "PlanningDone": true,
      "ProjectID": "string",
      "Reported": 0,
      "SpentManual": 0,
      "SpentOffline": 0,
      "SpentOnline": 0,
      "StartedBy": "string",
      "Status": "OPEN",
      "TrackerID": 0,
    },
    Object {
      "Active": false,
      "ActivityID": 0,
      "CloseComment": "string",
      "ClosedAt": 0,
      "ColdPlanningID": 0,
      "CreatedAt": 0,
      "Estimation": 0,
      "ID": 44,
      "IsColdPlanning": true,
      "IsManual": true,
      "IssueActivities": null,
      "IssueDone": 0,
      "IssueDueDate": 0,
      "IssueEstimation": 0,
      "IssueID": "string",
      "IssuePriorities": null,
      "IssueSpent": 0,
      "IssueStatuses": null,
      "IssueTitle": "string",
      "IssueTypes": null,
      "IssueURL": "string",
      "OpenComment": "string",
      "Outdated": true,
      "PlanningDone": true,
      "ProjectID": "string",
      "Reported": 0,
      "SpentManual": 0,
      "SpentOffline": 0,
      "SpentOnline": 0,
      "StartedBy": "string",
      "Status": "OPEN",
      "TrackerID": 0,
    },
  ],
  "stopPlanningError": Object {
    "code": 666,
    "message": "SOME_ERROR",
  },
  "timeSummary": Object {
    "sumManualMonth": 0,
    "sumManualToday": 0,
    "sumTotalDay": 0,
    "sumTotalMonth": 0,
  },
}
`;

exports[`[Store] Plannings Reducer [Success] [Snapshot] Should handle GetOpenPlannings 1`] = `
Object {
  "closedPlannings": Array [],
  "coldPlannings": Array [],
  "highlightedPlanning": "",
  "openPlannings": Array [
    Object {
      "Active": false,
      "ActivityID": 0,
      "CloseComment": "string",
      "ClosedAt": 0,
      "ColdPlanningID": 0,
      "CreatedAt": 0,
      "Estimation": 0,
      "ID": 0,
      "IsColdPlanning": true,
      "IsManual": true,
      "IssueActivities": null,
      "IssueDone": 0,
      "IssueDueDate": 0,
      "IssueEstimation": 0,
      "IssueID": "string",
      "IssuePriorities": null,
      "IssueSpent": 0,
      "IssueStatuses": null,
      "IssueTitle": "string",
      "IssueTypes": null,
      "IssueURL": "string",
      "OpenComment": "string",
      "Outdated": true,
      "PlanningDone": true,
      "ProjectID": "string",
      "Reported": 0,
      "SpentManual": 0,
      "SpentOffline": 0,
      "SpentOnline": 0,
      "StartedBy": "string",
      "Status": "OPEN",
      "TrackerID": 0,
    },
  ],
  "timeSummary": Object {
    "sumManualMonth": 0,
    "sumManualToday": 0,
    "sumTotalDay": 0,
    "sumTotalMonth": 0,
  },
}
`;

exports[`[Store] Plannings Reducer [Success] Should handle GetClosedPlannings 1`] = `
Object {
  "closedPlannings": Array [
    Object {
      "Active": false,
      "ActivityID": 0,
      "CloseComment": "string",
      "ClosedAt": 0,
      "ColdPlanningID": 0,
      "CreatedAt": 0,
      "Estimation": 0,
      "ID": 0,
      "IsColdPlanning": true,
      "IsManual": true,
      "IssueActivities": null,
      "IssueDone": 0,
      "IssueDueDate": 0,
      "IssueEstimation": 0,
      "IssueID": "string",
      "IssuePriorities": null,
      "IssueSpent": 0,
      "IssueStatuses": null,
      "IssueTitle": "string",
      "IssueTypes": null,
      "IssueURL": "string",
      "OpenComment": "string",
      "Outdated": true,
      "PlanningDone": true,
      "ProjectID": "string",
      "Reported": 0,
      "SpentManual": 0,
      "SpentOffline": 0,
      "SpentOnline": 0,
      "StartedBy": "string",
      "Status": "OPEN",
      "TrackerID": 0,
    },
  ],
  "coldPlannings": Array [],
  "highlightedPlanning": "",
  "openPlannings": Array [],
  "timeSummary": Object {
    "sumManualMonth": 0,
    "sumManualToday": 0,
    "sumTotalDay": 0,
    "sumTotalMonth": 0,
  },
}
`;

exports[`[Store] Plannings Reducer [Success] Should handle StartPlanning 1`] = `
Object {
  "closedPlannings": Array [],
  "coldPlannings": Array [],
  "highlightedPlanning": "",
  "openPlannings": Array [
    Object {
      "Active": true,
      "ActivityID": 0,
      "CloseComment": "string",
      "ClosedAt": 0,
      "ColdPlanningID": 0,
      "CreatedAt": 0,
      "Estimation": 0,
      "ID": 33,
      "IsColdPlanning": true,
      "IsManual": true,
      "IssueActivities": null,
      "IssueDone": 0,
      "IssueDueDate": 0,
      "IssueEstimation": 0,
      "IssueID": "string",
      "IssuePriorities": null,
      "IssueSpent": 0,
      "IssueStatuses": null,
      "IssueTitle": "string",
      "IssueTypes": null,
      "IssueURL": "string",
      "OpenComment": "string",
      "Outdated": true,
      "PlanningDone": true,
      "ProjectID": "string",
      "Reported": 0,
      "SpentManual": 0,
      "SpentOffline": 0,
      "SpentOnline": 0,
      "StartedBy": "string",
      "Status": "OPEN",
      "TrackerID": 0,
    },
    Object {
      "Active": false,
      "ActivityID": 0,
      "CloseComment": "string",
      "ClosedAt": 0,
      "ColdPlanningID": 0,
      "CreatedAt": 0,
      "Estimation": 0,
      "ID": 44,
      "IsColdPlanning": true,
      "IsManual": true,
      "IssueActivities": null,
      "IssueDone": 0,
      "IssueDueDate": 0,
      "IssueEstimation": 0,
      "IssueID": "string",
      "IssuePriorities": null,
      "IssueSpent": 0,
      "IssueStatuses": null,
      "IssueTitle": "string",
      "IssueTypes": null,
      "IssueURL": "string",
      "OpenComment": "string",
      "Outdated": true,
      "PlanningDone": true,
      "ProjectID": "string",
      "Reported": 0,
      "SpentManual": 0,
      "SpentOffline": 0,
      "SpentOnline": 0,
      "StartedBy": "string",
      "Status": "OPEN",
      "TrackerID": 0,
    },
  ],
  "timeSummary": Object {
    "sumManualMonth": 0,
    "sumManualToday": 0,
    "sumTotalDay": 0,
    "sumTotalMonth": 0,
  },
}
`;

exports[`[Store] Plannings Reducer [Success] Should handle StopPlanning 1`] = `
Object {
  "closedPlannings": Array [],
  "coldPlannings": Array [],
  "highlightedPlanning": "",
  "openPlannings": Array [
    Object {
      "Active": false,
      "ActivityID": 0,
      "CloseComment": "string",
      "ClosedAt": 0,
      "ColdPlanningID": 0,
      "CreatedAt": 0,
      "Estimation": 0,
      "ID": 33,
      "IsColdPlanning": true,
      "IsManual": true,
      "IssueActivities": null,
      "IssueDone": 0,
      "IssueDueDate": 0,
      "IssueEstimation": 0,
      "IssueID": "string",
      "IssuePriorities": null,
      "IssueSpent": 0,
      "IssueStatuses": null,
      "IssueTitle": "string",
      "IssueTypes": null,
      "IssueURL": "string",
      "OpenComment": "string",
      "Outdated": true,
      "PlanningDone": true,
      "ProjectID": "string",
      "Reported": 0,
      "SpentManual": 0,
      "SpentOffline": 0,
      "SpentOnline": 0,
      "StartedBy": "string",
      "Status": "OPEN",
      "TrackerID": 0,
    },
    Object {
      "Active": false,
      "ActivityID": 0,
      "CloseComment": "string",
      "ClosedAt": 0,
      "ColdPlanningID": 0,
      "CreatedAt": 0,
      "Estimation": 0,
      "ID": 44,
      "IsColdPlanning": true,
      "IsManual": true,
      "IssueActivities": null,
      "IssueDone": 0,
      "IssueDueDate": 0,
      "IssueEstimation": 0,
      "IssueID": "string",
      "IssuePriorities": null,
      "IssueSpent": 0,
      "IssueStatuses": null,
      "IssueTitle": "string",
      "IssueTypes": null,
      "IssueURL": "string",
      "OpenComment": "string",
      "Outdated": true,
      "PlanningDone": true,
      "ProjectID": "string",
      "Reported": 0,
      "SpentManual": 0,
      "SpentOffline": 0,
      "SpentOnline": 0,
      "StartedBy": "string",
      "Status": "OPEN",
      "TrackerID": 0,
    },
  ],
  "timeSummary": Object {
    "sumManualMonth": 0,
    "sumManualToday": 0,
    "sumTotalDay": 0,
    "sumTotalMonth": 0,
  },
}
`;

exports[`[Store] Plannings Reducer [SyncSuccess] Should handle TimeTickInActivePlanning 1`] = `
Object {
  "closedPlannings": Array [],
  "coldPlannings": Array [],
  "highlightedPlanning": "",
  "openPlannings": Array [
    Object {
      "Active": true,
      "ActivityID": 0,
      "CloseComment": "string",
      "ClosedAt": 0,
      "ColdPlanningID": 0,
      "CreatedAt": 0,
      "Estimation": 0,
      "ID": 33,
      "IsColdPlanning": true,
      "IsManual": true,
      "IssueActivities": null,
      "IssueDone": 0,
      "IssueDueDate": 0,
      "IssueEstimation": 0,
      "IssueID": "string",
      "IssuePriorities": null,
      "IssueSpent": 0,
      "IssueStatuses": null,
      "IssueTitle": "string",
      "IssueTypes": null,
      "IssueURL": "string",
      "OpenComment": "string",
      "Outdated": true,
      "PlanningDone": true,
      "ProjectID": "string",
      "Reported": 0,
      "SpentManual": 0,
      "SpentOffline": 0,
      "SpentOnline": 201.001,
      "StartedBy": "string",
      "Status": "OPEN",
      "TrackerID": 0,
    },
    Object {
      "Active": false,
      "ActivityID": 0,
      "CloseComment": "string",
      "ClosedAt": 0,
      "ColdPlanningID": 0,
      "CreatedAt": 0,
      "Estimation": 0,
      "ID": 44,
      "IsColdPlanning": true,
      "IsManual": true,
      "IssueActivities": null,
      "IssueDone": 0,
      "IssueDueDate": 0,
      "IssueEstimation": 0,
      "IssueID": "string",
      "IssuePriorities": null,
      "IssueSpent": 0,
      "IssueStatuses": null,
      "IssueTitle": "string",
      "IssueTypes": null,
      "IssueURL": "string",
      "OpenComment": "string",
      "Outdated": true,
      "PlanningDone": true,
      "ProjectID": "string",
      "Reported": 0,
      "SpentManual": 0,
      "SpentOffline": 0,
      "SpentOnline": 0,
      "StartedBy": "string",
      "Status": "OPEN",
      "TrackerID": 0,
    },
  ],
  "timeSummary": Object {
    "sumManualMonth": 0,
    "sumManualToday": 0,
    "sumTotalDay": 1.001,
    "sumTotalMonth": 1.001,
  },
}
`;
